source/tutorials/kubernetes.rst

```diff
@@ -3,170 +3,340 @@
+      being used for the pod networking. See `Choose a pod network add-on`_
+      for more information.
+
+   .. code:: bash
+
+      sudo kubeadm init \
+      --pod-network-cidr 10.244.0.0/16 #required for flannel
+
+
+.. warning:: 
+
+   Docker on |CL| will automatically use kata-runtime as the default Docker
+   runtime if it is available.
+
+   If you do not want to use the kata runtime or experience problems with
+   cluster initialization, you can disable Docker from setting kata as the
+   default runtime by running these commands:
+
+   .. code:: bash 
+
+      sudo rm /etc/systemd/system/docker.service.d/50-runtime.conf
+      sudo systemctl mask docker-set-default-runtime.service
+
+#. Once the cluster initialization is complete, continue reading about how to
+   `Use your cluster`_.
+
+containerd
+----------
+
+`containerd <https://containerd.io/>`_ is the runtime  that the Docker engine
+is built ontop of. Kubernetes can use containerd directly instead of going
+through the Docker engine for increased robustness and performance. See the
+`blog post on kubernetes containerd integration
+<https://kubernetes.io/blog/2018/05/24/kubernetes-containerd-integration-goes-ga/>`_
+for more details.
+
+#. Start the containerd service and enable it to run at boot automatically:
 
    .. code-block:: bash
 
-      sudo -s
+      sudo systemctl enable --now containerd.service
 
-#.  Create (or edit if it exists) the hosts file that Kubernetes will read to
-    locate the master's host:
 
-    .. code-block:: bash
+#. Configure kubelet to use containerd and reload the service.
 
-       echo "127.0.0.1 localhost `hostname`" >> /etc/hosts
+   .. code-block:: bash
 
-#.  Exit root:
+      sudo mkdir -p  /etc/systemd/system/kubelet.service.d/
 
-    .. code-block:: bash
+      cat << EOF | sudo tee  /etc/systemd/system/kubelet.service.d/0-containerd.conf
+      [Service]                                                 
+      Environment="KUBELET_EXTRA_ARGS=--container-runtime=remote --runtime-request-timeout=15m --container-runtime-endpoint=unix:///run/containerd/containerd.sock"
+      EOF
 
-       exit
+   .. code:: bash
 
-Configure and run Kubernetes
-****************************
+      systemctl daemon-reload
 
-This section describes how to configure and run Kubernetes with CRI-O and kata-runtime. We will be using  the :command:`kubectl` command-line interface to control Kubernetes.
+#. Run the kubeadm command to initialize the master node with the
+   :command:`--cri-socket` parameter:
 
-Configure and run CRI-O + kata-runtime
-======================================
+   .. important:: 
 
-#.  Enable the CRI-O service:
+      You may need to add additional parameters to the command below,
+      depending the pod network addon in use. In this example, *flannel* is
+      being used for the pod networking. See `Choose a pod network add-on`_
+      for more information.
```

Comment:
I think having two examples would be helpful here as well.

Reply:
Do you mean one example without flannel and one with? If so, that is one of the things I wanted to simplify. Examples trying to cover all the different cases is hard to follow.

Reply:
Yes I meant two example. Preferably the two most popular ones that are different would help.